examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,373 @@
+        # future.add_done_callback(log_metrics)
+        futures.append(future)
+
+    executor.shutdown(wait=True) # wait for processes and callbacks to complete 
+  
+    cuda_envs.close() 
+    cuda_envs.join_thread()
+    
+'''Run single instance of a finetuning job on given task. ''' 
+def train_finetune(task: str, save_ckpt: bool, load_path: str, wandb_group_name: str, save_folder: str, save_locally: bool, wandb_logger: Union[WandBLogger, None], bucket: Optional[str] = None, load_ignore_keys: Optional[List[str]] = None):
+    ft_hparams = TrainerHparams.create(cli_args=False, f=f'./composer/yamls/models/glue/{task}.yaml') 
+    ft_hparams.load_path = load_path
+    ft_hparams.device = DeviceGPU(torch.cuda.current_device())
+    ft_hparams.log_to_console = False 
+    ft_hparams.progress_bar = False
+    ft_hparams.load_ignore_keys = load_ignore_keys
+    
+    # add finetune-specific tags to the logger 
+    if wandb_logger:
+        wandb_logger._init_kwargs['tags'] = ['finetune', task]
+        wandb_logger._init_kwargs['group'] = wandb_group_name
+        wandb_logger._rank_zero_only = True
+        if not ft_hparams.loggers:
+            ft_hparams.loggers = []
+        ft_hparams.loggers.append(wandb_logger)
+
+    # load checkpoint to finetune on via cloud 
+    if not save_locally:
+        ft_hparams.load_object_store = S3ObjectStoreHparams(bucket=bucket)
+
+    # saving single checkpoint at the end of training the task 
+    if save_ckpt:
+        if save_locally:
+            task_ckpt_path = os.path.join(save_folder, f'{task}')
+            ft_hparams.save_folder = task_ckpt_path
+            ft_hparams.save_num_checkpoints_to_keep = 1
+            ft_hparams.save_overwrite = True
+
+            if not os.path.exists(task_ckpt_path):
+                os.mkdir(task_ckpt_path)
+        else:
+            if not ft_hparams.loggers:
+                ft_hparams.loggers = []
+            ft_hparams.loggers.append(ObjectStoreLogger(object_store_cls=S3ObjectStore, object_store_kwargs={'bucket': bucket}, use_procs=False))
+            ft_hparams.save_folder = f'{task}'
+            ft_hparams.save_num_checkpoints_to_keep = 0
+            save_artifact_name = f'{{run_name}}/{task}/ep{{epoch}}-ba{{batch}}-rank{{rank}}'
+            save_latest_artifact_name = f'{{run_name}}/{task}/latest-rank{{rank}}'
+            ft_hparams.save_artifact_name = save_artifact_name
+            ft_hparams.save_latest_artifact_name = save_latest_artifact_name
+            ft_hparams.save_overwrite = True
+
+    print(f"\n --------\n SPAWNING TASK {task.upper()}\n DEVICE: {torch.cuda.current_device()}\n --------")
+
+    trainer = ft_hparams.initialize_object()
+    trainer.fit()
+    print(f"\nFINISHED TRAINING TASK {task.upper()}\n")
+    return trainer.state.current_metrics
+
+'''Get the run name and S3 and WandB instances by pre-constructing a dummy TrainerHparams instance. ''' 
+def get_training_info():
```

Comment:
Curious why do we need this function? Similar to the above, would like to not presume that wandb or the object store logger is being used.